src/Searchlight/SearchlightEngine.cs

```diff
@@ -15,6 +16,11 @@ public class SearchlightEngine
         /// </summary>
         public List<SearchlightException> ModelErrors { get; } = new List<SearchlightException>();
 
+        /// <summary>
+        /// Represents the maximum size of a single page
+        /// </summary>
+        public static int MaximumPageSize { get; } = 1000;
```

Comment:
Hmm.  How does a developer change the maximum page size?  This makes it a constant.

It should probably say:

```
 public int MaximumPageSize { get; set; } = 1000;
```

Reply:
Actually, we should permit null:

```
public int? MaximumPageSize { get; set; }
```

Then, if the developer chooses null, we won't enforce maximum page size.